instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameGuesser.java

```diff
@@ -95,6 +96,21 @@ public Resource createResource(ConfigProperties config) {
         .orElseGet(Resource::empty);
   }
 
+  @Override
+  public boolean shouldApply(ConfigProperties config, Resource resource) {
+    String serviceName = config.getString("otel.service.name");
+    Map<String, String> resourceAttributes = config.getMap("otel.resource.attributes");
+    return serviceName == null
+        && !resourceAttributes.containsKey(ResourceAttributes.SERVICE_NAME.getKey())
+        && "unknown_service:java".equals(resource.getAttribute(ResourceAttributes.SERVICE_NAME));
```

Comment:
can you add a comment to explain that checking otel.service.name and otel.resource.attributes is just done for performance reasons (I think)?

Reply:
IIRC, It's not entirely for performance (although that's also true). This strategy is a "best guess" set of heuristics and we don't want to ever override a name applied by a previous strategy that might know better.

Reply:
I believe `otel.service.name` and `otel.resource.attributes` are applied after all of the user resource providers, so those would always take precedence, and so I think no need to check them here then?

Reply:
Makes sense, then this does indeed help to guard against attempting to load resources from disk and classpath.

Reply:
Added comment 👍